### PR TITLE
Temporary remove UAP target for DiagnosticSource

### PR DESCRIPTION
**Pull request by lmolkova**
DiagnosticSource nuget package preview should be published to nuget.org to be consumed by ApplicationInsights SDK (in preview).

On UWP, DiagnosticSource depends on Microsoft.NETCore.Platforms for UWP, that will not be published yet.

This change **temporary** removes UAP target from DiagnosticSource and removes dependency on it from System.Net.Http for UAP.

This change will be reverted once DiagnosticSource is build, published on MyGet and verified.

### CODE REVIEW

**Comment by ericstj** [comment]
Don't remove the configuration, if you do it will get stale.

If you want to exclude from the package just split up the configurations like we do for other projects.  EG: https://github.com/dotnet/corefx/blob/ccfae5c19c5f2aa5aa410302a9f29e837eb975a5/src/System.Numerics.Vectors/src/Configurations.props

**Comment by ericstj** [comment]
> On UWP, DiagnosticSource depends on Microsoft.NETCore.Platforms for UWP, that will not be published yet.

What do you mean by this?  Of course that package will be published.  It's our RID graph.  What's the actual dependency you have?

**Comment by vancem** [comment]
@ericstj - the ultimate goal here is simply to generate a 1.1.* (not 2.0) version of System.Net.Http and System.Diagnostics.DiagnosticSource nuget packages that can be published on Nuget.org.   Anything that achieves this is fine.   

The issue seems to be that we need to make sure that the package does not depend on any pre-release packages.   The delta was designed to do this, the idea being that we would merge it, get a build, and then put it back after we have the package we need (which we could publish to Nuget by hand).  

@lmolkova  @weshaggard @terrajobst  - It is not clear to me why we have to exclude the UAP (formally UWP).  There IS a version of Microsoft.NETCore.Platforms on Nuget that is not prerelease (1.1.0) right now, and I don't see why that would not 'just work'.   I suspect this is the heart of @ericstj issue.    Have we TRIED just using what is current built?  What is wrong with that?  

Anyway, given that we intend to revert the change shortly after doing it, it should not be a big deal one way or the other, but if we can do it without any modification, that would of course be better.  Why does that not work?

**Comment by vancem** [comment]
I defer to @ericstj and @weshaggard, but it seems like what we need is for the package to refer to the 1.1 versions of things would be the 'obvious' way of making this work.   Still given that we will revert, at this it may be easiest to simply use this PR as is.

File: src/System.Net.Http/src/netcore50/System/Net/HttpClientHandler.cs [hunk 1]
```diff
@@ -37,7 +37,6 @@ public class HttpClientHandler : HttpMessageHandler
 
         private readonly RTHttpBaseProtocolFilter rtFilter;
         private readonly HttpHandlerToFilter handlerToFilter;
-        private readonly HttpMessageHandler diagnosticsPipeline;
```
Comment (weshaggard) [inline src/System.Net.Http/src/netcore50/System/Net/HttpClientHandler.cs]: These changes look unrelated why are they included?

File: src/System.Net.Http/src/System.Net.Http.csproj [hunk 2]
```diff
@@ -117,10 +117,14 @@
       <Link>Common\System\Net\Logging\NetEventSource.Common.cs</Link>
     </Compile>
   </ItemGroup>
-  <!-- Common, except for Windows net46 build -->
```
Comment (weshaggard) [inline src/System.Net.Http/src/System.Net.Http.csproj]: We should need to change anything here.

File: src/System.Diagnostics.DiagnosticSource/src/Configurations.props [hunk 3]
```diff
@@ -4,7 +4,6 @@
     <BuildConfigurations>
```
Comment (weshaggard) [inline src/System.Diagnostics.DiagnosticSource/src/Configurations.props]: The only change you need to remove the uap dependency from the package is to change this file to:

```
    <PackageConfigurations>
      netstandard1.3;
      netstandard;
      netcoreapp;
      net45-Windows_NT;      
      net46-Windows_NT;
      netfx-Windows_NT;      
    </PackageConfigurations>
    <BuildConfigurations>
        $(PackageConfigurations);
        uap-Windows_NT;
    </BuildConfigurations>
```

That will still cause uap to build and be consumable in other places like Http but will exclude that configuration from your nuget package.

**Comment by weshaggard** [comment]
>the ultimate goal here is simply to generate a 1.1.* (not 2.0) version of System.Net.Http and System.Diagnostics.DiagnosticSource nuget packages that can be published on Nuget.org. Anything that achieves this is fine.

I was under the impression we are only publishing a System.Diagnostics.DiagnosticSouces package to push to nuget.org. This is the first I heard we want to do the same for System.Net.Http package from master. If that one is also needed then we will need to look at its dependencies independently to determine if there are any issues.  

The biggest thing we are trying to prevent is requiring us to ship a large set of our prerelease packages on nuget.org as they aren't ready for prime-time yet.

**Comment by lmolkova** [comment]
@ericstj  @weshaggard thanks for providing the solution with `PackageConfigurations`. I was not aware of this option.

@weshaggard the only package we are going to publish is DiagnosticSource, System.Net.Http will follow corefx schedule.

**Comment by weshaggard** [comment]
@lmolkova once we get a package build with your changes tomorrow morning I will have another look at the package to ensure the dependencies look correct.

**Comment by terrajobst** [comment]
> [@vancem] It is not clear to me why we have to exclude the UAP (formally UWP)

My motivation isn't to remove UAP support, it's removing a dependency on another preview package we are not willing to ship yet, specifically Microsoft.NETCore.Platforms.

**Comment by ericstj** [comment]
All you needed to do that is change the value for the $(PlatformPackageVersion) to point to the stable version: 1.1.0.

**Comment by lmolkova** [comment]
@weshaggard the package is finally built and available on myget. There is no UAP target. Could you please check it?
